### DISCUSSION TITLE
CefSharp.Core.Runtime.dll The specified module could not be found

### DISCUSSION THREAD

**Discussion by Ravnholt2102** [cefsharp/CefSharp#4733] [General]
I have a very simple Offscreen console application.
It check if Cef.IsInitialized and if not it runs Cef.InitializeAsync().

When i run this, i get this exception:
System.IO.FileNotFoundException: 'Could not load file or assembly 'C:\Sources\PreviewGenerators\ConsoleGenerator\ConsoleGenerator\bin\x64\Debug\net8.0\runtimes\win-x64\lib\netcoreapp3.1\CefSharp.Core.Runtime.dll'. The specified module could not be found.'

If i replace InitializeAsync() with Initialize() it works as expected.

I have installed nuget CefSharp.OffScreen.NETCore 121.3.130
Using .NET 8.0

**Comment by amaitland** [comment]
Impossible to say what's going on without an example.

Does the problem reproduce with the `MinimalExample`? It currently targets `.Net 6.0`, you can easily change that to `.Net 8`

https://github.com/cefsharp/CefSharp.MinimalExample/blob/master/CefSharp.MinimalExample.OffScreen/Program.cs#L57

--- 

Specifying a `RuntimeIdentifier` will likely resolve your issue.

https://github.com/cefsharp/CefSharp/blob/master/NuGet/PackageReference/Readme.txt#L10

**Reply by Kation** [reply]
.Net 6.0 is fine. Change to .Net 8.0 failed.

**Reply by Kation** [reply]
oh, use `net8.0-windows` is fine.

**Comment by Ravnholt2102** [comment]
I have added an example here: https://github.com/Ravnholt2102/ConsoleTest
I think this is an issue on my end, since this is such a simple example that completely fails.

**Reply by amaitland** [reply]
Have you considered https://github.com/hardkoded/puppeteer-sharp

If you have to use `CefSharp.OffScreen`, then you'll need to use a `SynchronizationContext` to ensure the calls  run on the correct threads.

https://github.com/cefsharp/CefSharp.MinimalExample/blob/master/CefSharp.MinimalExample.OffScreen/Program.cs#L40C41-L40C63
